crates/storage-query-datafusion/src/remote_query_scanner_manager.rs

```diff
@@ -106,15 +126,38 @@ impl RemoteScannerManager {
     pub fn get_partition_target_node(
         &self,
         table: &str,
-        _partition_id: PartitionId,
-    ) -> PartitionLocation {
-        // TODO: obtain this information from the metadata, once exposed
-        PartitionLocation::Local {
-            scanner: self
-                .local_partition_scanner(table)
-                .expect("should be present"),
+        partition_id: PartitionId,
+    ) -> anyhow::Result<PartitionLocation> {
+        let my_node_id = self.my_node_id();
+
+        match self.partition_routing.get_node_by_partition(partition_id) {
+            None => {
+                self.partition_routing.request_refresh();
+                bail!("node lookup for partition {} failed", partition_id)
```

Comment:
I am wondering whether retrying once after the refresh has completed would give a noticeable better experience. While this can't catch all problems it will catch the common problem where there was a change in the cluster but not all nodes have realized this. Now when a query comes in, we are still operating with outdated information which could be solved by refreshing the routing information.

Reply:
I think my comment does not make a lot of sense here. Ignore it.